ProcessMaker/processmaker
# FOUR-8338 | Add Pagination to Templates List in Select Template Modal

Ticket: [FOUR-8338](https://processmaker.atlassian.net/browse/FOUR-8338)

# Issue
We need to improve the Template list in the Create Process Modal by adding pagination. Currently, the modal is set to show up to 300 templates (so that users can see all templates, assuming they have 300 or less templates created).

[Pagination Mock-Up and Specs](https://processmaker.atlassian.net/wiki/spaces/PM4/pages/2586050576/Pagination)

# Solution
Add Vue Bootstrap Paginaton to Template Search component.

# Updated UI
![Screen Shot 2023-05-15 at 10 46 51 AM](https://github.com/ProcessMaker/processmaker/assets/73713665/93a11ce7-5794-4c96-8327-492676dd18b9)

# How to Test
1. Go to branch `task/FOUR-8338` in Core.
2. In the ProcessMaker UI, go to Designer -> Processes -> Create New Process.
3. In the Select Template Modal, at the bottom of the modal, view the pagination.
	- There are 18 templates listed per page.
	- Pagination should accurately reflect the number of templates in your environment.

# Related Tickets & Packages
- [FOUR-8311](https://processmaker.atlassian.net/browse/FOUR-8311) | Review and Adjust Limitation in Process Creation Menu to Show Only First 10 Templates
- [FOUR-7375](https://processmaker.atlassian.net/browse/FOUR-7375) | Process Templates - Spring 2023

# Code Review Checklist


[FOUR-8338]: https://processmaker.atlassian.net/browse/FOUR-8338?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FOUR-8311]: https://processmaker.atlassian.net/browse/FOUR-8311?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

.
